# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/ridaeman02/flyrank-ml-internship/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

I select **Lane 2: Refresh / Content Opportunity Scoring**.

Many web pages lose search traffic over time (content decay). In our dataset, more than half of the pages are declining. Editors need to know which important pages to update first. This lane helps them find those pages using search and engagement data.

In [2]:
import pandas as pd
df = pd.read_csv("../../data/raw/content_refresh_anonymized.csv")
print("Trend direction breakdown (percentage):")
print(df['trend_direction'].value_counts(normalize=True) * 100)

Trend direction breakdown (percentage):
trend_direction
down      54.206667
stable    19.873333
up        14.626667
new        7.453333
flat       3.840000
Name: proportion, dtype: float64


## 2. The question: decision, action, cost of a wrong call

- **Decision**: Which declining pages should the content team refresh or expand first?
- **Who acts**: Content editors and writers will update or rewrite the recommended pages.
- **Cost of a wrong call**: If we recommend the wrong page (false positive), the editor wastes hours updating a page that did not need it. If we miss a declining page (false negative), we lose search traffic and revenue.
- **Why ML**: We have 30,000 pages across 32 clients. There are many mixed signals like clicks, impressions, page age, and scroll rates. A simple rule is not enough to rank them all efficiently.

In [4]:
print("Number of pages per client (showing first 5):")
print(df['client_id'].value_counts().head(5))

Number of pages per client (showing first 5):
client_id
client_19581e27de    7008
client_6208ef0f77    3681
client_4e07408562    2294
client_3fdba35f04    2267
client_f369cb89fc    1796
Name: count, dtype: int64


## 3. Quick look at the data (2-3 real numbers)

Here are 3 real numbers from our starter dataset:
1. **Total size**: 30,000 pages across 32 different clients.
2. **Decline rate**: 54.21% of all pages have a downward trend (`trend_direction == 'down'`).
3. **Average page age**: The average page age is 256.2 days, which shows many pages might be stale.

In [6]:
print(f"Total pages: {len(df)}")
print(f"Unique clients: {df['client_id'].nunique()}")
print(f"Declining pages: {(df['trend_direction'] == 'down').mean() * 100:.2f}%")
print(f"Average page age: {df['content_age_days'].mean():.1f} days")

Total pages: 30000
Unique clients: 32
Declining pages: 54.21%
Average page age: 256.2 days


## 4. Careful words: what I can and can't claim

- **What we can claim**: We provide **decision-support** recommendations and **directional** rankings based on **observed** historical search visibility, position, and engagement patterns to prioritize editor review.
- **What we cannot claim**: We cannot make **causal proof** claims (e.g., "refreshing this page will cause traffic to double") because we are using observational historical data, not running randomized controlled trials. We cannot "predict Google's algorithm."

In [8]:
print("Claims check: Observed, directional, decision-support claims only.")

Claims check: Observed, directional, decision-support claims only.


## Self-check

Before you submit, confirm each line honestly:

- [x] Every section above is filled — markdown thinking AND the code that backs it
- [x] The notebook runs top to bottom with no errors (Runtime → Run all)
- [x] No client names, URLs, or private queries anywhere
- [x] My claims use careful words: observed, measured, directional, decision-support
- [x] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.